en/test_and_ci/continous_integration.md

```diff
@@ -1,3 +1,98 @@
+
+In addition, if there are changes to the `main` branch Jenkins builds / deploys metadata used by QGroundControl and documentation builds, based on the definitions in [PX4-Autopilot/Jenkinsfile](https://github.com/PX4/PX4-Autopilot/blob/main/Jenkinsfile).
+
+
+## PX4 Documentation
+
+Jenkins uses the [PX4BuildBot](https://github.com/PX4BuildBot) account to upload the following metadata to the [PX4-user_guide](https://github.com/PX4/PX4-user_guide) repo.
+
+- `airframes_reference.md`: Markdown for [Airframes Reference Documentation](../airframes/airframe_reference.md)
+- `parameter_reference.md`: Markdown for [Parameter Reference Documentation](../advanced_config/parameter_reference.md)
+- `modules/*.md`: Markdown for [Modules documentation](../modules/README.md)
+- `graph_*.json`: JSON defining uORB Graph, visualized in [uORB Graph](../middleware/uorb_graph.md) page
+- `msg_docs/*.md`: Markdown for [uORB Messages documentation](../msg_docs/README.md)
+- `failsafe_sim/*`: Files for [Failsafe Simulator Interface](../config/safety_simulation.md)
+
+An example commit can be found [here](https://github.com/PX4/PX4-user_guide/commit/a4ffaf5fc32df8650da614aa3129b3b90794584f).
+
+PX4 documentation is built separately using a separate Jenkins process following the above update (or any other other updates).
+The PX4 CI build process is defined in: [PX4-user_guide/Jenkinsfile](https://github.com/PX4/PX4-user_guide/blob/main/Jenkinsfile).
+
+
+## QGroundControl 
+
+The PX4 firmware uploaded by `deploy_all.yml` is used by QGroundControl for [PX4 Firmware updates](../config/firmware.md) (stable: `release/*` / beta: `main`).
+ 
+The Jenkins server uses the [PX4BuildBot](https://github.com/PX4BuildBot) account to upload the following metadata to QGC whenever there is an update to main or a release branch:
+
+- `airframes.xml`: Airframes metadata describing all supported airframes
+- `parameters.xml`: Parameters metadata describing all supported parameters
+
+QGC uses the above information for populating the [airframe selection](../config/airframe.md) page for PX4 and the _default_ parameter metadata information (QGC will use parameter metadata from the vehicle instead if it is available).
+An example commit of parameter metadata can be found [here](https://github.com/mavlink/qgroundcontrol/commit/7f4f3b6253fe80a881e9a91a1f5b6d960ad11834).
+
+Note that QGroundControl normally gets metadata for [Events](../concept/events_interface.md#implementation)and [Parameters](../advanced/parameters_and_configurations.md#publishing-parameter-metadata-to-a-gcs) from the vehicle (it is built into the firmware, or a URL is provided in the firmware from which it can be downloaded).
+
+
+## Other
+
+Jenkins also updates the PX4 ROS messages repository [PX4/px4_msgs](https://github.com/PX4/px4_msgs) and uploads the metadata to the [Amazon S3 bucket server](https://px4-travis.s3.amazonaws.com/).
```

Comment:
"uploads the metadata to the [Amazon S3 bucket server](https://px4-travis.s3.amazonaws.com/)." - what metadata? YOu already covered a lot here, is this metadata something associated with px4_msgs?

Reply:
Ooh I think so, I didn't cover it specifically but can add that info as well

Reply:
If it is metadata associated with ROS 2  messages then "probably not". The only dependency now from PX4 in ROS2 is the PX4/px4_msgs. There used to be more stuff, but XRCE-DDS means it isn't needed any more.